### PR TITLE
Add Service for System Analytics

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
# Fixes or Implements [AB#8775](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8775)

* [x] Enhancement
* [ ] Bug
* [ ] Documentation

## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"? 

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

## Description

Adds the System Analytics service that converts the database models into csv streams.

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [x] Not Required

### Steps to Reproduce:

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

NO

### Browsers Tested

* [ ] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.

### CODE REVIEW

File: Apps/AdminWebClient/src/Server/Services/CsvExportService.cs [hunk 1]
```diff
@@ -17,46 +17,80 @@ namespace HealthGateway.Admin.Services
...
+        private const int PageSize = 100000;
+        private const int Page = 0;
+        private readonly INoteDelegate noteDelegate;
+        private readonly IUserProfileDelegate userProfileDelegate;
+        private readonly ICommentDelegate commentDelegate;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="CsvExportService"/> class.
         /// </summary>
-        /// <param name="logger">Injected Logger Provider.</param>
```
Comment (sslaws) [inline Apps/AdminWebClient/src/Server/Services/CsvExportService.cs:34]: NIT:  We should not be removing the logger and should in general be adding debug statements at least.

File: Apps/AdminWebClient/src/Server/Services/CsvExportService.cs [hunk 2]
```diff
@@ -17,46 +17,80 @@ namespace HealthGateway.Admin.Services
...
-            return retStream;
+            DBResult<IEnumerable<UserProfile>> profiles = this.userProfileDelegate.GetAll(Page, PageSize);
+            return GetStream<UserProfile, UserProfileCsvMap>(profiles.Payload);
+        }
+
+        private static Stream GetStream<TModel, TMap>(IEnumerable<TModel> obj)
+            where TMap : ClassMap
+        {
+            MemoryStream stream = new MemoryStream();
+            using (StreamWriter writeFile = new StreamWriter(stream, leaveOpen: true))
```
Comment (sslaws) [inline Apps/AdminWebClient/src/Server/Services/CsvExportService.cs:82]: Did you test this?  I found when the StreamWriter closed it closed the underlying stream and the controller didn't work.

File: Apps/AdminWebClient/src/AdminWebClient.csproj [hunk 3]
```diff
@@ -18,9 +18,9 @@
   </PropertyGroup>  
 
   <ItemGroup>
+    <PackageReference Include="CsvHelper" Version="15.0.5" />
     <PackageReference Include="Microsoft.AspNetCore.Authentication.OpenIdConnect" Version="3.1.6" />
     <PackageReference Include="Microsoft.AspNetCore.SpaServices.Extensions" Version="3.1.6" />
-    <PackageReference Include="ServiceStack.Text" Version="5.9.2" />
```
Comment (leobcgov) [inline Apps/AdminWebClient/src/AdminWebClient.csproj:23]: Just in case you have removed this by accident.

Comment (sslaws) [reply Apps/AdminWebClient/src/AdminWebClient.csproj:23]: I introduced and Tiago didn't use it.  Removed to keep code tidy.

File: Apps/AdminWebClient/src/Server/Services/CsvExportService.cs (hunk 1 above)
Comment (FuriousLlama) [reply Apps/AdminWebClient/src/Server/Services/CsvExportService.cs:34]: +1

File: Apps/AdminWebClient/src/Server/Services/CsvExportService.cs (hunk 2 above)
Comment (tiagograf-AOT) [reply Apps/AdminWebClient/src/Server/Services/CsvExportService.cs:82]: I tested and it worked

**Pull request merged**